packages/react/src/components/FormGroup/FormGroup.stories.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import FormGroup from '../FormGroup';
+import TextInput from '../TextInput';
+import RadioButtonGroup from '../RadioButtonGroup';
+import RadioButton from '../RadioButton';
+import Button from '../Button';
+import { Stack } from '../Stack';
+
+export default {
+  title: 'Components/FormGroup',
+  component: FormGroup,
```

Comment:
One pattern we've started to follow is that if a prop does not impact the component visuals at all, we hide/remove it from the controls panel. Also if a prop does not have a meaningful control (ones with type of node, for instance like `children`) should also be hidden/removed from the controls panel.

Putting these up here at the component-level makes these apply to all stories, instead of just one. So these will be hidden from the controls pane for all stories. They will still show up in the component docs tables though on the `Docs` tab!

```suggestion
  component: FormGroup,
  argTypes: {
    legendId: {
      table: {
        disable: true,
      },
    },
    className: {
      table: {
        disable: true,
      },
    },
    children: {
      table: {
        disable: true,
      },
    },
  },
```

Reply:
> One pattern we've started to follow is that if a prop does not impact the component visuals at all, we hide/remove it from the controls panel. Also if a prop does not have a meaningful control (ones with type of node, for instance like `children`) should also be hidden/removed from the controls panel.
> 
> Putting these up here at the component-level makes these apply to all stories, instead of just one. So these will be hidden from the controls pane for all stories. They will still show up in the component docs tables though on the `Docs` tab!

I wont lie, i see these in a bunch of components

```
export default {
  title: 'Components/FormGroup',
  component: FormGroup,
  parameters: {
    docs: {
      page: mdx,
    },
  },
};
```

and even ones that have `subComponents` such as 

```
export default {
  title: 'components/TreeView',
  component: TreeView,
  subcomponents: {
    TreeNode,
  },
  args: {},
  argTypes: {
    children: {
      table: {
        disable: true,
      },
    },

    className: {
      table: {
        disable: true,
      },
    },

    label: {
      table: {
        disable: true,
      },
    },

    onSelect: {
      table: {
        disable: true,
      },
    },
  },
};
```

i dont actually know how this works or what it does. I undrstand that, default will be the default thing in the storybook, and im slowlylearning passing arg types and args pass in controls, but why set it up this way vs just exporting a component like others do, or what or how does  `subcomponents` work or what is it for?

maybe if i understand this better I can understand these stories better

Reply:
> argTypes: {

to also add to more questions than above

```
argTypes: {
    legendId: {
      table: {
        disable: true,
      },
    },
    className: {
      table: {
        disable: true,
      },
    },
    children: {
      table: {
        disable: true,
      },
    },
  },
```

setting the arg types with the prop name followed by table than disabled, that is a storybook thing to tell all components within this particular components storybook, to disable them from showing up?

Reply:
@tay1orjones  please see the above, thank you

Reply:
@TannerS Yeah, I know it's confusing. I will say reading through the storybook docs can help, but I'll annotate one of those examples with some descriptions:

```js
export default {
  title: 'components/TreeView', // Place all stories in this file in a folder "TreeView" inside the section "components"
  component: TreeView, // Adds this component to the API/prop tables in the "Docs" tab
  subcomponents: {
    TreeNode, // Also adds this component to the API/prop tables in the "Docs" tab
  },
  args: {}, // Define "component level" args that will apply to every story in this file
  argTypes: { // Define "component level" argTypes that will apply to every story in this file
    children: {
      table: {
        disable: true,
      },
    },
```

---

> setting the arg types with the prop name followed by table than disabled, that is a storybook thing to tell all components within this particular components storybook, to disable them from showing up?

Yes, exactly. It tells storybook that for every story in this component/file, `disable` the prop from appearing in the controls `table`

Hope this helps!